### PR TITLE
Fix typos in configuration docs

### PR DESCRIPTION
**Pull request by joel-hamill**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -404,8 +404,8 @@
         "<code>org.apache.kafka.common.serialization.Serde</code> interface.";
 
     public static final String WINDOWED_INNER_CLASS_SERDE = "windowed.inner.class.serde";
-    private static final String WINDOWED_INNER_CLASS_SERDE_DOC = " Default serializer / deserializer for the inner class of a windowed record. Must implement the \" +\n" +
-        "        \"<code>org.apache.kafka.common.serialization.Serde</code> interface.. Note that setting this config in KafkaStreams application would result " +
+    private static final String WINDOWED_INNER_CLASS_SERDE_DOC = " Default serializer / deserializer for the inner class of a windowed record. Must implement the " +
+        "<code>org.apache.kafka.common.serialization.Serde</code> interface. Note that setting this config in KafkaStreams application would result " +
```
Comment (showuon) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:408]: Nice catch! Note, current doc is like this:
image

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 2]
```diff
@@ -216,8 +216,10 @@
...
-                                                                            + " If conflicting configurations are set and idempotence is not explicitly enabled, idempotence is disabled.";
+                                                                            + " Additionally, enabling idempotence requires the value of this configuration to be less than or equal to " + MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_FOR_IDEMPOTENCE + "."
+                                                                            + " If conflicting configurations are set and idempotence is not explicitly enabled, idempotence is disabled. "
+                                                                            + " Record ordering is preserved when <code>enable.idempotence</code> is set to true for idempotent "
+                                                                            + " producer (or transactional producer), even when max in-flight requests are greater than 1 (supported up to 5).";
```
Comment (showuon) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: Actually, I think the original version is better. Could you revert this change, and only fix the typo in stream config?

**Review by showuon — COMMENTED** [review]
@joel-hamill , thanks for the PR. Left some comment.

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java (hunk 2 above)
Comment (wsun-confluent) [reply clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: Hello and thanks for reviewing!  The callout about record ordering was originally requested within Confluent; the same is mentioned at this page: https://developer.confluent.io/tutorials/message-ordering/kafka.html.  We think this would also benefit AK docs.

**Review by showuon — APPROVED** [review]
LGTM!

**Comment by joel-hamill** [comment]
cc @ijuma

**Review by andrewchoi5 — APPROVED** [review]

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java (hunk 2 above)
Comment (C0urante) [reply clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: The last sentence is redundant though, isn't it? The docs already state:

> Note that if this config is set to be greater than 1 and <code>enable.idempotence</code> is set to false, there is a risk of message re-ordering after a failed send due to retries (i.e., if retries are enabled).

It's fine if we want to clarify that re-ordering is not a risk when `enable.idempotence` is set to true, but we should also try to keep the docs here concise.

What about adding a brief followup in that sentence instead?

> Note that if this config is set to be greater than 1 and <code>enable.idempotence</code> is set to false, there is a risk of message re-ordering after a failed send due to retries (i.e., if retries are enabled); ordering will be preserved if retries are disabled and/or <code>enable.idempotence</code> is true.

Comment (joel-hamill) [reply clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: @C0urante i have updated the text with your feedback. thanks.

**Review by guozhangwang — APPROVED** [review]
LGTM!
